numpy/__init__.pxd

```diff
@@ -917,8 +917,8 @@ cdef extern from "numpy/ufuncobject.h":
     int PyUFunc_ReplaceLoopBySignature \
         (ufunc, PyUFuncGenericFunction, int *, PyUFuncGenericFunction *)
     object PyUFunc_FromFuncAndDataAndSignature \
-             (PyUFuncGenericFunction *, void **, char *, int, int, int,
-              int, char *, char *, int, char *)
+             (PyUFuncGenericFunction *, void *const *, const char *, int, int, int,
+              int, const char *, const char *, int, const char *)
```

Comment:
Quick comment.  Thanks, I bet the other changes are all great.  This may have been intentional, though.
Cython is very pedantic about const mismatches, so I am not sure we do anyone a favor by fixing this here.

Reply:
I didn't intentionally leave this out in #23847; it was an oversight on my part. But I would have been ignorant of the Cython subtleties. What would this break for Cython? How do we test it?

If you would prefer, I can just undo this part of the diff.

Reply:
Let's at least split it out, since I think everything else is internal.

Cython matches signatures exactly, and I am not even sure it allows to easily cast between them, so you force everyone with working code to update (and it might be impossible to do so and also support older versions).

Reply:
Done.